bindings/node/package.json

```diff
@@ -67,7 +66,12 @@
     "mongodb": ">=3.4.0"
   },
   "engines": {
-    "node": ">=6"
+    "node": ">=12.9.0"
+  },
+  "binary": {
+    "napi_versions": [
+      4
```

Comment:
Should this be 6? Also "binary" doesn't appear to be an npm field, just curious what uses this

Reply:
prebuild told me to do this :) https://www.npmjs.com/package/prebuild#n-api-considerations

Reply:
Also, eh, I’m not sure. 4 is what we settled on – having the instance data feature available is *much* nicer though, and as the comments in the source indicate, it is available in all Node.js versions that have N-API version 4 (just not as stable yet – the feature has not been changed since it was experimental, so essentially we can consider it stable from the beginning)

Reply:
I think that's fine to rely on, but just so I understand in the header file we have:
```
#define NAPI_VERSION 6
#define NAPI_EXPERIMENTAL
```
should it be `NAPI_VERSION 4` and keep the define `NAPI_EXPERIMENTAL` instead? Essentially we're saying compiling against 4 with the experimental instance data feature would be the same as 6 but we still want to target 4. IIUC

And related to prebuild it says we should throw a define in the [node-gyp file](https://www.npmjs.com/package/prebuild#node-gyp), but since we're already successfully compiling this seems like a best practice thing?

Reply:
> should it be `NAPI_VERSION 4` and keep the define `NAPI_EXPERIMENTAL` instead? Essentially we're saying compiling against 4 with the experimental instance data feature would be the same as 6 but we still want to target 4. IIUC

The problem is that in current Node.js versions, where the feature is no longer considered experimental, it’s only available with `NAPI_VERSION >= 6`.

> And related to prebuild it says we should throw a define in the [node-gyp file](https://www.npmjs.com/package/prebuild#node-gyp), but since we're already successfully compiling this seems like a best practice thing?

That’s for defining the `NAPI_VERSION` in the source code – it’s what these defines here are already doing :)

Reply:
Oh ok I see, thanks for clarifying. And yea I also saw that can be used to conditionally change the NAP_VERSION so, no need!